src/caffeine/caffeine.c

```diff
@@ -21,20 +21,28 @@ static gex_Rank_t rank, size;
   const int double_Complex_workaround =4100;
 #endif
 
-void caf_caffeinate(mspace* symmetric_heap, intptr_t* symmetric_heap_start, gex_TM_t* initial_team)
+void caf_caffeinate(mspace* symmetric_heap, intptr_t* symmetric_heap_start, mspace* non_symmetric_heap, gex_TM_t* initial_team)
 {
   GASNET_SAFE(gex_Client_Init(&myclient, &myep, initial_team, "caffeine", NULL, NULL, 0));
 
-  size_t segsz = GASNET_PAGESIZE; // TODO: how big can we make this?
+  size_t segsz = GASNET_PAGESIZE*10; // TODO: how big can we make this?
 
   gex_Segment_t mysegment;
   GASNET_SAFE(gex_Segment_Attach(&mysegment, *initial_team, segsz));
-  // TODO: split into symmetric and non-symmetric heaps
+
   *symmetric_heap_start = (intptr_t)gex_Segment_QueryAddr(mysegment);
+  size_t total_heap_size = gex_Segment_QuerySize(mysegment);
+  float non_symmetric_fraction = 0.5f; // TODO: Configurable via env var?
+  size_t symmetric_heap_size = total_heap_size * non_symmetric_fraction;
+  size_t non_symmetric_heap_size = total_heap_size - symmetric_heap_size;
```

Comment:
Pretty sure the two lines above are inverted from your intent, based on variable naming

Reply:
Woops. Good catch.